core/sdk-core/src/main/java/software/amazon/awssdk/core/client/handler/BaseAsyncClientHandler.java

```diff
@@ -73,7 +75,12 @@ protected BaseAsyncClientHandler(SdkClientConfiguration clientConfiguration,
         ClientExecutionParams<InputT, OutputT> executionParams,
         AsyncResponseTransformer<OutputT, ReturnT> asyncResponseTransformer) {
 
+        // For streaming requests, prepare() should be called as early as possible to avoid NPE in client
+        // See https://github.com/aws/aws-sdk-java-v2/issues/1268
+        CompletableFuture<ReturnT> asyncTransformerFuture = asyncResponseTransformer.prepare();
```

Comment:
can we write a unit test for it to verify it's being called so that we know if it accidentally gets removed in the future?

Reply:
Added